### PR TITLE
Task/uf permissions refactor

### PR DESCRIPTION
**Pull request by nwang92**
I made the UF depend on splunk_common, so all roles are now controlled by an individual main.yml within roles/splunk_common/tasks/main.yml

Custom image with these plays have passed tests, but I think this says more of our tests than of this change :)
```
$ pytest -sv tests/test_debian_9.py
================= test session starts =================
platform darwin -- Python 2.7.15, pytest-4.0.2, py-1.7.0, pluggy-0.8.0 -- /Users/nwang/github/docker-splunk/venv/bin/python2.7
cachedir: .pytest_cache
rootdir: /Users/nwang/github/docker-splunk, inifile:
collected 23 items

tests/test_debian_9.py::TestDebian9::test_splunk_entrypoint_help PASSED
tests/test_debian_9.py::TestDebian9::test_splunk_entrypoint_create_defaults PASSED
tests/test_debian_9.py::TestDebian9::test_splunk_entrypoint_start_no_password PASSED
tests/test_debian_9.py::TestDebian9::test_splunk_entrypoint_start_no_accept_license PASSED
tests/test_debian_9.py::TestDebian9::test_uf_entrypoint_help PASSED
tests/test_debian_9.py::TestDebian9::test_uf_entrypoint_create_defaults PASSED
tests/test_debian_9.py::TestDebian9::test_uf_entrypoint_start_no_password PASSED
tests/test_debian_9.py::TestDebian9::test_uf_entrypoint_start_no_accept_license PASSED
tests/test_debian_9.py::TestDebian9::test_adhoc_1so_using_default_yml PASSED
tests/test_debian_9.py::TestDebian9::test_adhoc_1so_using_default_yml_hec_ssl_disabled PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_trial PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_custombuild PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_namedvolumes PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_command_start PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_command_start_service PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_java_oracle PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_java_openjdk PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_hec PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1so_apps PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1uf_hec PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1uf_apps PASSED
tests/test_debian_9.py::TestDebian9::test_compose_1uf1so PASSED
tests/test_debian_9.py::TestDebian9::test_compose_2idx2sh PASSED

================= warnings summary =================
tests/test_debian_9.py::TestDebian9::test_adhoc_1so_using_default_yml
  /Users/nwang/github/docker-splunk/venv/lib/python2.7/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
    InsecureRequestWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================= 23 passed, 1 warnings in 915.91 seconds =================
```

### CODE REVIEW

File: roles/splunk_universal_forwarder/tasks/main.yml [hunk 1]
```diff
@@ -1,50 +1,32 @@
...
-    option: "{{ item.opt }}"
-    value: "{{ item.val }}"
-  with_items:
-    - { opt: 'USERNAME', val: 'admin' }
-    - { opt: 'PASSWORD', val: '{{ splunk.password }}' }
-  loop_control:
-    label: "{{ item.opt }}"
-  no_log: "{{ hide_password }}"
-
-- include_tasks: pre_splunk_start_commands.yml
```
Comment (jmeixensperger) [inline roles/splunk_universal_forwarder/tasks/main.yml]: Is this play still required for forwarders? Looks like they are not executing those tasks anymore.

Comment (nwang92) [reply roles/splunk_universal_forwarder/tasks/main.yml]: I moved this to the common role, so technically all roles can utilize the `SPLUNK_BEFORE_START_CMD` env to run something before Splunk starts

**Comment by lephino** [comment]
pull in master, so circle ci reruns

**Comment by nwang92** [comment]
@lephino We don't have CircleCI setup for splunk-ansible yet

**Pull request merged**